warp10/src/main/java/io/warp10/standalone/StandaloneStoreClient.java

```diff
@@ -387,45 +387,37 @@ public boolean hasNext() {
             Entry<byte[], byte[]> kv = iterator.peekNext();
 
             // If the next key is over the range, nullify startrow unless we still have boundaries to fetch
-            if (0 == preBoundary && 0 == postBoundary && Bytes.compareTo(kv.getKey(), stoprow) > 0) {
-              startrow = null;
-            } else {
+            if (!(0 == preBoundary && 0 == postBoundary && Bytes.compareTo(kv.getKey(), stoprow) > 0)) {
               //
               // If we are time based or value count based with values left to read or with boundaries still to be fetched,
               // return true
               if (-1 == fcount || nvalues > 0 || preBoundary > 0 || postBoundary > 0) {
                 return true;
-              } else {
-                startrow = null;
               }
             }
-          } else {
-            startrow = null;
           }
           
-          // We need to reseek if startrow is null (it indicates we need to skip to the next GTS)
-          if (null == startrow) {
-            idx++;
+          // We need to reseek because we need to skip to the next GTS
```

Comment:
Leave a comment indicating this is determined by startrow having been set to null